ConvivaIntegration/components/conviva/ConvivaAnalyticsTask.brs

```diff
@@ -150,27 +153,60 @@ end sub
 sub onAdBreakStarted()
   m.LivePass.detachStreamer()
   m.LivePass.adStart()
+
+  adBreak = m.top.player.callFunc(m.top.player.BitmovinFunctions.AD_LIST)[m.PodIndex]
+  duration = 0
+
+  for each ad in adBreak.ads
+    duration += ad.duration
+  end for
+
+  if adBreak.scheduleTime = 0
```

Comment:
Please extract this to a method. Web name of this method `mapAdPosition(event.adBreak, this.player)`